backoff_test.go

```diff
@@ -0,0 +1,11 @@
+package grpc
+
+import "testing"
+
+func TestBackoffConfigDefaults(t *testing.T) {
+	b := BackoffConfig{}
+	b.setDefaults()
+	if b != *DefaultBackoffConfig {
```

Comment:
I think DefaultBackoffConfig should be a constant instead of var (in backoff.go). It should not be mutable. Also making it a pointer sounds weird to me.